typescript/packages/lattice-client-core/src/lattice-client.ts

```diff
@@ -199,9 +195,36 @@ export class LatticeClient {
   /**
    * Connect to the lattice
    */
-  readonly #connect = async (): Promise<void> => {
-    await this.#connection?.connect();
-  };
+  async #connect(count = 0): Promise<void> {
```

Comment:
we can use `await` like normal here right, since it's `async` -- was that version actually less readable than this one?

Also, it *might* be overkill, but I would probably want that `retryCount` (currently named `count`) to be in some sort of `ConnectArgs`/`ConnectOptions` object -- there's no way it will only have a retry count for long!

Reply:
Yes, we sure can—thanks! Way easier to read now, too.